### DISCUSSION TITLE
Processing previously forwarded messages in the source channel

### DISCUSSION THREAD

**Discussion by User087** [aahnik/tgcf#232] [Q&A] [answered]
I find that when a message in the source channel is edited, the same message in the destination channel also gets edited, and when a message in the source channel is replied to with some message, the same message in the destination channel is also replied to with that message. It seems the relevant code for this is in tgcf/live.py, but line 90 onwards there seems to also indicate that when a message gets deleted from the source channel then the same message should get deleted from the destination channel, but I find that isn't the case. Is there a bug or have I misinterpreted that part of the code?

I've also found that when the bot is turned off and started up again, this behaviour no longer occurs for messages that were forwarded before the bot was restarted, so if those messages are edited or replied to then they just come out as new messages. Could you explain how this behaviour works in the bot? How is it able to remember which message in the destination channel corresponds to the one in the source channel that is being edited/replied to? Is it stored in some temporary memory that gets deleted when the bot is restarted? Presumably it doesn't involve storing the details of the forwarded message in some external file as I find this behaviour occurring even when the bot is deployed on Heroku.

It might also be worth mentioning this behaviour in the README or Wiki, as not all forwarder bots have this behaviour.

**Comment by aahnik** [comment, marked as answer]
> deleted from the source channel then the same message should get deleted from the destination channel

You need to turn on delete sync in your configuration. read the [config schema in wiki](https://github.com/aahnik/tgcf/wiki/How-to-configure-tgcf-%3F#schema)

even if you turn on delete sync, the feature does not work in some cases, as telegram does not always notify the client when a message is deleted. i find that delete sync does not work well with bot accounts

> It might also be worth mentioning this behaviour in the README or Wiki, as not all forwarder bots have this behaviour.

Very good suggestion! I will do that.

> I've also found that when the bot is turned off and started up again ...

The data is stored in memory, and gets lost on stopping. I will be adding persistence. So that people who want can save the data of past messages. In future versions, you will also be able to configure how many past messages or how many MB of data is stored.

**Reply by aahnik** [reply]
hi @User087 , please star and share this repo!

**Reply by User087** [reply]
> The data is stored in memory, and gets lost on stopping. I will be adding persistence. So that people who want can save the data of past messages. In future versions, you will also be able to configure how many past messages or how many MB of data is stored.

Would this work even on Heroku? If so then how? Where would the data be saved, considering their ephemeral file system?

**Reply by aahnik** [reply]
I am trying to solve #201 and #202 

> Would this work even on Heroku? If so then how? 

It would require a connection to an external file storage service. Maybe a database, or Amazon S3, or GitHub Gist. 
Or maybe store in Telegram's Saved Messages.

I have not implemented anything yet. I am trying to use something that is free and easy for deployers.
